src/abacus-ts/selectors/account.ts

```diff
@@ -0,0 +1,172 @@
+import { getCurrentMarketId } from '@/state/perpetualsSelectors';
+
+import { calculateFills } from '../calculators/fills';
+import {
+  calculateAllOrders,
+  calculateOpenOrders,
+  calculateOrderHistory,
+} from '../calculators/orders';
+import {
+  calculateMarketsNeededForSubaccount,
+  calculateParentSubaccountPositions,
+  calculateParentSubaccountSummary,
+} from '../calculators/subaccount';
+import { calculateTransfers } from '../calculators/transfers';
+import { mergeLoadableStatus } from '../lib/mapLoadable';
+import {
+  selectRawFillsLiveData,
+  selectRawFillsRest,
+  selectRawFillsRestData,
+  selectRawIndexerHeight,
+  selectRawIndexerHeightData,
+  selectRawMarkets,
+  selectRawMarketsData,
+  selectRawOrdersLiveData,
+  selectRawOrdersRest,
+  selectRawOrdersRestData,
+  selectRawParentSubaccount,
+  selectRawParentSubaccountData,
+  selectRawTransfersLiveData,
+  selectRawTransfersRest,
+  selectRawTransfersRestData,
+  selectRawValidatorHeight,
+  selectRawValidatorHeightData,
+} from './base';
+
+const selectRelevantMarketsList = createAppSelector(
+  [selectRawParentSubaccountData],
+  (parentSubaccount) => {
+    if (parentSubaccount == null) {
+      return undefined;
+    }
+    return calculateMarketsNeededForSubaccount(parentSubaccount);
+  }
+);
+
+const selectRelevantMarketsData = createAppSelector(
+  [selectRelevantMarketsList, selectRawMarketsData],
+  (marketIds, markets) => {
+    if (markets == null || marketIds == null) {
+      return undefined;
+    }
+    return pick(markets, ...marketIds);
+  },
+  {
+    // use shallow equal for result so that we only update when these specific keys differ
+    memoizeOptions: { resultEqualityCheck: shallowEqual },
+  }
+);
+
+export const selectParentSubaccountSummary = createAppSelector(
+  [selectRawParentSubaccountData, selectRelevantMarketsData],
+  (parentSubaccount, markets) => {
+    if (parentSubaccount == null || markets == null) {
+      return undefined;
+    }
+    const result = calculateParentSubaccountSummary(parentSubaccount, markets);
+    return result;
+  }
+);
+
+export const selectParentSubaccountPositions = createAppSelector(
+  [selectRawParentSubaccountData, selectRelevantMarketsData],
+  (parentSubaccount, markets) => {
+    if (parentSubaccount == null || markets == null) {
+      return undefined;
+    }
+    return calculateParentSubaccountPositions(parentSubaccount, markets);
+  }
+);
+
+export const selectParentSubaccountSummaryLoading = createAppSelector(
+  [selectRawParentSubaccount, selectRawMarkets],
+  mergeLoadableStatus
+);
+
+export const selectParentSubaccountOpenPositions = createAppSelector(
+  [selectParentSubaccountPositions],
+  (positions) => {
+    return positions?.filter((p) => p.status === IndexerPerpetualPositionStatus.OPEN);
+  }
+);
+export const selectParentSubaccountOpenPositionsLoading = selectParentSubaccountSummaryLoading;
+
+const baseTime = { height: 0, time: '1971-01-01T00:00:00Z' };
```

Comment:
nit: upper case const and maybe pull to top of file, `baseTime` -> `BASE_TIME`

Reply:
good call